src/cpp/shared/scripting/callback.h

```diff
@@ -21,14 +21,11 @@ void ironfist_lua_pushmulti();
 
 /*
  * The implementations use Lua functions like lua_isnumber and lua_tonumber.
- * We need template specialization to implement these.
+ * We need template specialization to implement these, see callback.cpp.
  */
 template <typename Res>
 nonstd::optional<Res> PopLuaResult(lua_State *L, int arg);
 
-template <>
-nonstd::optional<bool> PopLuaResult(lua_State *L, int arg);
```

Comment:
I'm really surprised that template specializations don't need to be in the header file